compiler/frontend/src/org/jetbrains/kotlin/resolve/DescriptorResolver.java

```diff
@@ -1094,6 +1094,18 @@ scopeForDeclarationResolution, container, false, new TraceBasedLocalRedeclaratio
                     substitutedSuperType = approximatingSuperType;
                 }
 
+                UnwrappedType unwrapped = type.unwrap();
+                boolean lowerNullable = FlexibleTypesKt.lowerIfFlexible(unwrapped).isMarkedNullable();
+                boolean upperNullable = FlexibleTypesKt.upperIfFlexible(unwrapped).isMarkedNullable();
+                if (lowerNullable != upperNullable) {
+                    return KotlinTypeFactory.flexibleType(
```

Comment:
Sorry for being tedious, but speaking formally this is a language semantics change, thus it would be better to hide it behind the language feature, too

Reply:
Done. I gave this case a slightly different error message since platform types are non-denotable, but it may still not be very clear.